### PR TITLE
Correctly manage infra status condition when there is no LB controller

### PR DESCRIPTION
**Pull request by alvaroaleman**
We currently only update the infra status after we got an event. If we
reconcile a hostedcluster that uses an LB for the apiserver in a
management cluster that does not support those, we will never update the
status because we never get an event.

Recheck after a minute when the infrastructure is not ready to make sure
the condition gets updated regularly.

/cc @csrwng 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller_test.go [hunk 1]
```diff
@@ -693,8 +694,7 @@ func TestEtcdRestoredCondition(t *testing.T) {
 	}
 }
 
-func TestEventHandling(t *testing.T) {
-	t.Parallel()
+func sampleHCP() (*hyperv1.HostedControlPlane, error) {
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller_test.go:697]: is there a reason to return an error? should we just panic?

Comment (alvaroaleman) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller_test.go]: Updated it to take a testing.T and just Fatal if it fails

**Comment by alvaroaleman** [comment]
@csrwng updated, ptal

**Pull request merged**